fdmi/plugins/motr-fdmi-app.md

```diff
@@ -1,33 +1,88 @@
+In addition, this interface inherits the horizontal scalability property from MOTR while being reliable since it will be transactionally coupled with the core.
+  
+FDMI applications can be seen as traditional object or key-value applications that has a more closed relationship with CORTX-Motr. 
 
-`./fdmi_plugin_st.sh`
+## FDMI Implementations Overview
 
-This shell script starts Motr services using a specific filter configuration, starts the fdmi_plugin_sample application to listen for specific records that match with the filter already setup and do some key-value operations that reflects this specifications. 
+FDMI is a scalable publish-subscribe interface where each Motr instance works as a source and produces records that describe operations. 
 
-The output will show whether the test was successful or not. 
+On the other side, there is a FDMI Application or Plugin that registers a filter in the Motr Filter Database to selects records that match such filters.
 
-A second way to test this fdmi_sample_plugin will be by directly running the fdmi_sample app wrapper with the following command:
+The filter substrings to match the `Value` part of the index Key/Value parir are configured at the CORTX initialization time.
 
-`./fdmi_app`
+The FDMI source instances send matched records to the FDMI Application or Plugin in batches with their transactional contexts.
 
-This method will need to have a Cortx-motr cluster already up and running beforehand that includes FDMI capabilities by adding a fdmi_filters section in the CDF.yaml config file. 
-You can check if your cluster is running with FDMI filter capability by looking at the /etc/motr/confd.xc file in which case will have the specific filter-id for the fdmi_sample_plugin. 
+For each record, the FDMI Application or Plugin performs actions and sends acknowledgements back to the source instances using Motr RPC calls.
 
-Example of such record in /etc/motr/confd.xc file will be:
+## How to create and FDMI Application and run with examples
 
-' {0x6c| ((^l|1:81), 2, ^l|1:81, "", ^n|1:3, ^v|1:66, [1: "Bucket-Name"], [1: "192.168.12.2@tcp:12345:4:1"])},'
+All FDMI related service code and examples are in the `cortx-motr/fdmi` directory. 
 
-More details about the FDMI design and settings can be found in this link: 
+One of the FDMI Application example is within the `plugins` directory named as fdmi_sample_plugin. 
+
+This application is written in C using `motr/client.h` interface and it adds the new MOTR client to the cluster and starts listening for FDMI records or events that match the FDMI filter substrings.  
+
+The executable will be compiled as part of the initial Motr compilation using the Makefile.am dependency and the Makefile.sub file.
+
+This program then communicates with the fdmi_app python wrapper by printing to stdout the records. 
+
+To test this application we need to run the fdmi_app wrapper typing in the console the following command:
+
+`sudo ./fdmi_app`
```

Comment:
Then, all you did in the previous step was launch the listener. Correct?

In which case, there are more instructions required so that the reader can test their listener.

Reply:
I fixed this so that the text actually reflects that the fdmi_app is just a python script to test the fdmi_sample_plugin.